decidim-core/app/models/decidim/participatory_process_step.rb

```diff
@@ -11,5 +11,25 @@ class ParticipatoryProcessStep < ApplicationRecord
     validates :end_date, date: { after: :start_date, allow_blank: true, if: proc { |obj| obj.start_date.present? } }
 
     validates :active, uniqueness: { scope: :decidim_participatory_process_id }, if: proc { |step| step.active? }
+
+    validates :position, numericality: { greater_than_or_equal_to: 0, only_integer: true }, allow_blank: true
+
+    before_create :set_position
+
+    private
+
+    # Internal: Sets the position of the step if it is `nil`. That means that
+    # when the step is the first of the process, and no position is set
+    # manually, it will be set to 0, and when it is not the only one it will be
+    # set to the last step's position + 1.
+    #
+    # Note: This allows manual positioning, so it can cause different steps to
+    # have the same position.
+    def set_position
+      return if position.present?
+      return self.position = 0 if participatory_process.steps.empty?
+
+      self.position = participatory_process.steps.last.position + 1
```

Comment:
What about `participatory_process.steps.pluck(:position).last` ? (I know we'll load more rows, but we'll load less data, not sure which is best though.)